tests/unittests/tests-senml_json/tests-senml_json.c

```diff
@@ -0,0 +1,563 @@
+                                "\"t\":1.276020091e+09,\"v\":23.6}]";
+
+char senml_json_baseinfo[]    = "[{\"bn\":\"urn:dev:ow:10e2073a01080063\",\"u\":\"Cel\","
+                                "\"t\":1.276020076e+09,\"v\":23.5},{\"u\":\"Cel\","
+                                "\"t\":1.276020091e+09,\"v\":23.6}]";
+
+char senml_json_updatetime[]  = "[{\"n\":\"John\",\"u\":\"V\",\"v\":120.398,\"t\":123456789,"
+                                "\"ut\":5}]";
+
+char senml_json_stringvalue[] = "[{\"n\":\"Jackie\",\"vs\":\"what a nice string\","
+                                "\"t\":123456790}]";
+
+char senml_json_basevalues[]  = "[{\"bver\":5,\"bn\":\"riot_device\",\"bt\":123456000,"
+                                "\"bu\":\"Cel\",\"bv\":-273.15,\"bs\":89.12,\"n\":\"Hans\","
+                                "\"vs\":\"what a nice string\",\"t\":123456790}]";
+
+char senml_json_datavalue[]   = "[{\"bver\":5,\"bn\":\"riot_device\",\"bt\":123456000,"
+                                "\"bu\":\"Cel\",\"bv\":-54.0001,\"n\":\"Kim\",\"t\":-10,"
+                                "\"ut\":12,\"vd\":\"U2VuTUwgaXMgc28gY29vbCEK\"}]";
+
+char senml_json_valuesum[]    = "[{\"n\":\"Donald\",\"u\":\"W\",\"s\":1240.812,\"t\":987654321}]";
+
+char senml_json_boolvalue[]   = "[{\"n\":\"Daisy\",\"vb\":true}]";
+
+char senml_json_link[]        = "[{\"n\":\"Link\",\"l\":\"[{\\\"href\\\":\\\"derp\\\","
+                                "\\\"foo\\\":\\\"bar\\\"}]\",\"vb\":true}]";
+
+char expected_with_baseinfo[] = "[{\"bver\":5,\"bn\":\"urn:dev:mac:0024befffe804ff2;\","
+                                "\"bt\":1468342153.341000,\"bu\":\"Cel\",\"bv\":21.000000,"
+                                "\"bs\":8237823.412000,"
+                                "\"n\":\"temperature\",\"t\":-5.000000,\"ut\":5.000000,"
+                                "\"v\":0.500000},{\"n\":\"temperature\",\"t\":-10.000000,"
+                                "\"ut\":5.000000,\"v\":0.700000}]";
+
+char expected_no_baseinfo[]   = "[{\"n\":\"float_measure\",\"u\":\"m\",\"t\":1480778196.431000,"
+                                "\"ut\":10.000000,\"v\":42.195000},{\"n\":\"with_sum\","
+                                "\"u\":\"m\",\"t\":1480778906.541000,\"ut\":10.000000,"
+                                "\"s\":1423576.331000,\"v\":39.190000},{\"n\":\"string_measure\","
+                                "\"t\":1480778388.934000,\"ut\":30.000000,\"vs\":\"hello i am a "
+                                "string value\"},{\"n\":\"bool_measure\",\"t\":1480778500.871000,"
+                                "\"ut\":1.000000,\"vb\":false},{\"n\":\"data_measure\","
+                                "\"l\":\"[{\\\"href\\\":\\\"derp\\\",\\\"foo\\\":"
+                                "\\\"bar\\\"}]\",\"t\":1480778629.230000,\"ut\":120.000000,"
+                                "\"vd\":\"VGhpcyBpcyBiaW5hcnkgdmFsdWUK\"}]";
+
+char output[512];
```

Comment:
`static`, yes, but not `const`. `senml_decode_json_s` inserts `'\0'` after a string value and a `char *` in a record will point to a position in the input string. Otherwise, strings would have to be copied, and I felt like that would be too ugly without malloc and waste memory.